### PR TITLE
Updated Readme + Setup.py for PyPa

### PR DESCRIPTION
**Pull request by story645**
## PR Summary
PyPI will  render readmes written in [plaintext, markdown or rst](https://packaging.python.org/guides/making-a-pypi-friendly-readme/) , so wanted to take advantage of that since the current mpl page is kind of sparse: https://pypi.org/project/matplotlib/

While updating the readme & setup.py for pictures (logo and a grainy version of the frontpage quartet - would love some help here to improve it), I also added the discourse and funding links in setup.py ([project urls are fluid](https://packaging.python.org/guides/distributing-packages-using-setuptools/#project-urls) & updated some of the readme links 'cause the old references had been moved. Also shortened and moved around some content to try and make the readme a bit concise. 

This is also sort of a model for what packages should do if #16592 goes in.

### CODE REVIEW

**Comment by tacaswell** [comment]
Is there anyway to check how pypi is going to interpret the updated strings (short of releasing as seeing what it does)?

File: README.rst [hunk 1]
```diff
@@ -40,46 +40,45 @@
...
 
-NOTE: The current master branch is now Python 3 only.  Python 2 support is
-being dropped.
+Matplotlib produces publication-quality figures in a variety of hardcopy formats
+and interactive environments across platforms. Matplotlib can be used in Python scripts, 
+the Python and IPython shell, web application servers, and various
+graphical user interface toolkits.
 
-`Home page <http://matplotlib.org/>`_
+**Note:** The master branch is Python 3 only.  Python 2 support is being dropped.
```
Comment (tacaswell) [inline README.rst]:
```suggestion
**Note:** The master branch is Python 3 only.  Python 2 support has been dropped.
```

**Comment by story645** [comment]
@tacaswell apparently yes: https://packaging.python.org/guides/making-a-pypi-friendly-readme/#validating-restructuredtext-markup
```
twine check dist/*
```

File: README.rst [hunk 2]
```diff
@@ -40,46 +40,45 @@
...
 
-NOTE: The current master branch is now Python 3 only.  Python 2 support is
-being dropped.
+Matplotlib produces publication-quality figures in a variety of hardcopy formats
+and interactive environments across platforms. Matplotlib can be used in Python scripts, 
+the Python and IPython shell, web application servers, and various
+graphical user interface toolkits.
 
-`Home page <http://matplotlib.org/>`_
+**Note:** The master branch is Python 3 only.  Python 2 support has been dropped.
```
Comment (timhoffm) [inline README.rst]: I don‘t think we need this anymore. It‘s 2020 and the default assumption should be Python 3.

File: README.rst [hunk 3]
```diff
@@ -40,46 +40,45 @@
...
+`install <http://matplotlib.org/users/installing.html>`_ documentation. 
 
 Test
 ====
 
-After installation, you can launch the test suite::
+After installation, launch the test suite::
 
-  pytest
+  python -mpytest
```
Comment (timhoffm) [inline README.rst]:
```suggestion
  python -m pytest
```

While technically correct, not using a space may confuse people less familiar with bash. We‘ve changed these also some time ago in the docs.

**Comment by tacaswell** [comment]
I'm going to merge this and then manually do the backport so I can (finally) get the 3.2.0 tag done.

**Comment by tacaswell** [comment]
I have done the backport locally, will push soon.

**Comment by tacaswell** [comment]
Manually backported via 65a35ef

**Pull request merged**